### PR TITLE
Add tag categories

### PR DESCRIPTION
**Pull request by chrisjohntan**
This should:
- Create tag categories as an enum class
- Allow tag colour to be retrieved from a tag object

### CODE REVIEW

File: src/main/java/seedu/address/model/tag/Tag.java [hunk 1]
```diff
@@ -33,6 +36,32 @@ public static boolean isValidTagName(String test) {
         return test.matches(VALIDATION_REGEX);
     }
 
+    /**
+     * Set the tag category.
+     * @param tagCategory
+     */
+    public void setTagCategory(TagCategory tagCategory) {
```
Comment (chrisjohntan) [inline src/main/java/seedu/address/model/tag/Tag.java:43]: Should tagCategory be passed as String or TagCategory object?

File: src/main/java/seedu/address/model/tag/Tag.java [hunk 2]
```diff
@@ -14,16 +14,19 @@ public class Tag {
     public static final String VALIDATION_REGEX = "\\p{Alnum}+(-\\p{Alnum}+)*";
 
     public final String tagName;
+    private TagCategory tagCategory;
```
Comment (yooplo) [inline src/main/java/seedu/address/model/tag/Tag.java:17]: This should help to distinguish which category the tag object belongs to

File: src/main/java/seedu/address/model/tag/Tag.java (hunk 1 above)
Comment (blackpanther9229) [reply src/main/java/seedu/address/model/tag/Tag.java:43]: I prefer TagCategory object

Comment (yooplo) [reply src/main/java/seedu/address/model/tag/Tag.java:43]: I would feel TagCategory Object would  be a better practice for abstraction.

File: src/main/java/seedu/address/model/tag/Tag.java [hunk 3]
```diff
@@ -33,6 +36,32 @@ public static boolean isValidTagName(String test) {
...
+     */
+    public TagCategory getTagCategory() {
+        return this.tagCategory;
+    }
+
+    /**
+     * Returns the colour code corresponding to the tag category.
+     * @return String containing the hex code for the colour.
+     */
+    public String getTagColour() {
```
Comment (yooplo) [inline src/main/java/seedu/address/model/tag/Tag.java:59]: Good to be able to retrieve the color for the UI component.

File: src/main/java/seedu/address/model/tag/TagCategory.java
Comment (yooplo) [inline src/main/java/seedu/address/model/tag/TagCategory.java:1]: Yup this works.

**Review by yooplo — APPROVED** [review]
Looks Good, this should effectively allow each tag to store the additional information of which category they belong to.

**Pull request merged**